### PR TITLE
[MOBDW-89] - UI ux/storybook

### PR DESCRIPTION
**Pull request by BrianCorbin**
Soundtrack of this PR: [Opening To Shrek](https://www.youtube.com/watch?v=w2jFpdWSOvc)

### Motivation

UI Component testing is great so you don't have to start the app and navigate to the view every time you make a minor UI tweak :)

### In this PR

Integrated Storybook into the project repo

### Caveats

None

### Future Work

Need to create all of the Stories for all of the Views

[MOBDW-99]

### Testing

Test Suites: 25 passed, 25 total
Tests:       108 passed, 108 total
Snapshots:   8 passed, 8 total
Time:        100.274 s
Ran all test suites.
✨  Done in 105.40s.

[MOBDW-99]: https://mobilecoin.atlassian.net/browse/MOBDW-99

### CODE REVIEW

**Comment by BrianCorbin** [comment]
Oh, all of the commits from [PR#49](https://github.com/mobilecoinofficial/desktop-wallet/pull/49) are captured here. I must have branched from there instead of main. My bad! Won't cause any issues, but just a few extra things in the file changes.

File: app/components/TransactionInfoLabel/TransactionInfoLabel.tsx [hunk 1]
```diff
@@ -3,11 +3,11 @@ import type { FC } from 'react';
...
 import { makeStyles, Typography } from '@material-ui/core';
 
-import type { Theme } from '../theme';
+import type { Theme } from '../../theme';
 
-interface TransactionInfoLabelProps {
+export interface TransactionInfoLabelProps {
   amount?: number;
-  sign: string;
+  sign: '+' | '-';
```
Comment (fktt22) [inline app/components/TransactionInfoLabel/TransactionInfoLabel.tsx:10]: Doesn't ESLint require props to be in alphabetical order?

**Review by fktt22 — CHANGES_REQUESTED** [review]
There is an empty translation.missing.json file

Stories should be in the same directory as the components they relate to
(This also applies to tests; they will be moved accordingly too)

**Review by fktt22 — COMMENTED** [review]
Just remove the empty file
I'm not sure about leaving the examples for building stories

File: app/components/TransactionInfoLabel/TransactionInfoLabel.tsx (hunk 1 above)
Comment (BrianCorbin) [reply app/components/TransactionInfoLabel/TransactionInfoLabel.tsx:10]: They should be, are they not?

**Comment by davidyhall** [comment]
It looks like this branch works off of another. We should get into the habit of working off of main.

**Review by davidyhall — CHANGES_REQUESTED** [review]
We should add to the README how to run storybook.

I'm getting these errors

```
➜  desktop-wallet git:(ui-ux/storybook_MOBDW-89) ✗ yarn storybook

yarn run v1.22.4
$ start-storybook -p 6006
info @storybook/react v6.1.21
info
ERR! Error: No configuration files have been found in your configDir (/Users/davidyhall/Desktop/desktop-wallet/.storybook).
ERR! Storybook needs either a "main" or "config" file.
ERR!     at validateConfigurationFiles (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/utils/validate-configuration-files.js:50:11)
ERR!     at loadCustomPresets (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/common/custom-presets.js:19:43)
ERR!     at loadConfig (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/config.js:66:139)
ERR!     at startPreview (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/dev-server.js:408:49)
ERR!     at storybookDevServer (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/dev-server.js:505:48)
ERR!     at async buildDevStandalone (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-dev.js:317:33)
ERR!     at async buildDev (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-dev.js:380:3)
ERR!  Error: No configuration files have been found in your configDir (/Users/davidyhall/Desktop/desktop-wallet/.storybook).
ERR! Storybook needs either a "main" or "config" file.
ERR!     at validateConfigurationFiles (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/utils/validate-configuration-files.js:50:11)
ERR!     at loadCustomPresets (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/common/custom-presets.js:19:43)
ERR!     at loadConfig (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/config.js:66:139)
ERR!     at startPreview (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/dev-server.js:408:49)
ERR!     at storybookDevServer (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/dev-server.js:505:48)
ERR!     at async buildDevStandalone (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-dev.js:317:33)
ERR!     at async buildDev (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-dev.js:380:3)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

^C
➜  desktop-wallet git:(ui-ux/storybook_MOBDW-89) ✗ yarn build-storybook
yarn run v1.22.4
$ build-storybook
info @storybook/react v6.1.21
info
info => Cleaning outputDir /Users/davidyhall/Desktop/desktop-wallet/storybook-static
info => Building manager..
info => Loading manager config..
ERR! Error: No configuration files have been found in your configDir (/Users/davidyhall/Desktop/desktop-wallet/.storybook).
ERR! Storybook needs either a "main" or "config" file.
ERR!     at validateConfigurationFiles (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/utils/validate-configuration-files.js:50:11)
ERR!     at loadCustomPresets (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/common/custom-presets.js:19:43)
ERR!     at loadConfig (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/manager/manager-config.js:198:143)
ERR!     at buildManager (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-static.js:201:56)
ERR!     at buildStaticStandalone (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-static.js:270:11)
ERR!  Error: No configuration files have been found in your configDir (/Users/davidyhall/Desktop/desktop-wallet/.storybook).
ERR! Storybook needs either a "main" or "config" file.
ERR!     at validateConfigurationFiles (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/utils/validate-configuration-files.js:50:11)
ERR!     at loadCustomPresets (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/common/custom-presets.js:19:43)
ERR!     at loadConfig (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/manager/manager-config.js:198:143)
ERR!     at buildManager (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-static.js:201:56)
ERR!     at buildStaticStandalone (/Users/davidyhall/Desktop/desktop-wallet/node_modules/@storybook/core/dist/server/build-static.js:270:11)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by BrianCorbin** [comment]
> It looks like this branch works off of another. We should get into the habit of working off of main.

Yeah, I meant to work off of main and must have misclicked the branch in my git GUI application. Will be more diligent of that in the future :)

**Comment by BrianCorbin** [comment]
> We should add to the README how to run storybook.

The correct command should be `yarn run storybook`

I'll work on a README for it, will that be necessary for this commit?

**Comment by BrianCorbin** [comment]
Oh huh, it looks like git never started tracking the main.js storybook file, that's weird. Explains why you're getting those errors. Lemme fix that rn.

**Comment by BrianCorbin** [comment]
Hah, yeah, so the .gitignore file was auto ignoring the main.js storybook file, lol. Fixed with next commit.

**Review by davidyhall — APPROVED** [review]
LGTM!

**Comment by davidyhall** [comment]
Rebased to drop the other branch's commits. Tests should also run now too. Will merge after passes!

**Comment by davidyhall** [comment]
We reintroduced the other branch into this PR to just move faster. It all looks good and test are passing locally!

**Pull request merged**